### PR TITLE
Fix context Id and make loader work with request scope

### PR DESCRIPTION
**Pull request by tarazou9**
Fix context Id and make loader work with request scope

### CODE REVIEW

**Comment by tarazou9** [comment]
@krislefeber  Would you please take a look at this and let me know? :)

**Comment by krislefeber** [comment]
@tarazou9 , I'm not sure I understand what feature your changes provide. It appears to be functionally equivalent to how it was before your changes. Am I missing something? I see that you have the context id, but it's never being used. In fact, the previous code was already using the request object to tie the dataloader object to, so it was in fact request scoped

File: index.ts [hunk 1]
```diff
@@ -43,19 +44,21 @@ export class DataLoaderInterceptor implements NestInterceptor {
...
-        if (ctx[type] === undefined) {
-          try {           
-            const contextId = ContextIdFactory.getByRequest(ctx.req);         
-            ctx[type] = (await this.moduleRef
-              .resolve<NestDataLoader<any, any>>(type, contextId, { strict: false }))
-              .generateDataLoader();
-          } catch (e) {
-            throw new InternalServerErrorException(`The loader ${type} is not provided` + e);
+      ctx[NEST_LOADER_CONTEXT_KEY] = {
+        contextId: ContextIdFactory.create(),
```
Comment (krislefeber) [inline index.ts]: contextId is never used.

Comment (lennykean) [reply index.ts:48]: `contextId` is used on line 53

**Comment by lennykean** [comment]
@krislefeber The data loader is forced into singleton scope from a DI perspective because of the use of `ModuleRef.get`. While it *does* get attached to the request, it must first be resolved, and with the current implementation, that can't happen unless the dataloader and its DI subtree is all singleton scope.

For example, if your `NestDataLoader<>` service has a dependency `Foo` that is request scoped, `Foo` will fail to resolve because the call to `get` can't locate a request scoped service. @tarazou9's fix is switching it to a `resolve`. That also means `DataLoaderInterceptor` must create and store a contextId.

**Pull request merged**